packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -188,6 +211,34 @@ export async function produceBlockBody<T extends BlockType>(
           if (payload.transactions.length === 0) {
             this.metrics?.blockPayload.emptyPayloads.inc({prepType});
           }
+
+          // Capella and later forks have withdrawals on their ExecutionPayload
+          if (fork === ForkName.capella || fork === ForkName.eip4844) {
+            // TODO EIP-4844 Remove this when the EC includes `withdrawals`
+            (blockBody as capella.BeaconBlockBody).executionPayload.withdrawals = [];
+          }
+
+          if (fork === ForkName.eip4844) {
+            // SPEC: https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/validator.md#blob-kzg-commitments
+            // After retrieving the execution payload from the execution engine as specified in Bellatrix, use the
+            // payload_id to retrieve blobs and blob_kzg_commitments via get_blobs_and_kzg_commitments(payload_id)
+            // TODO EIP-4844: getBlobsBundle and getPayload must be either coupled or called in parallel to save time.
+            const blobsBundle = await this.executionEngine.getBlobsBundle(payloadId);
+
+            // Sanity check consistency between getPayload() and getBlobsBundle()
+            const blockHash = toHex(payload.blockHash);
```

Comment:
here we return payload block hash but seems later on we get by beacon block hash

Reply:
Get by execution payload blockHash

https://github.com/ChainSafe/lodestar/blob/95739bde2cc380ab0089910f0a22eac51e63be21/packages/beacon-node/src/chain/chain.ts#L424-L425

Set by execution payload blockHash

https://github.com/ChainSafe/lodestar/blob/95739bde2cc380ab0089910f0a22eac51e63be21/packages/beacon-node/src/chain/chain.ts#L397

Which is computed here from the payload

https://github.com/ChainSafe/lodestar/blob/95739bde2cc380ab0089910f0a22eac51e63be21/packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts#L229